src/libs/ascent/ascent.cpp

```diff
@@ -77,6 +77,91 @@ check_for_file(const std::string &file_name,
 
 }
 
+//-----------------------------------------------------------------------------
+void
+load_included_files_in_node_tree(conduit::Node &node, int mpi_comm_id)
+{
+    // This function recursively traverses the node tree searching for include statements
+    // If one is found, the node attempts to load the file's contents and add it to the node tree
+    int comm_size = 1;
+    int rank = 0;
+
+#ifdef ASCENT_MPI_ENABLED
+    if(mpi_comm_id == -1)
+    {
+      // do nothing, an error will be thrown later
+      // so we can respect the exception handling
+      return;
+    }
+    MPI_Comm mpi_comm = MPI_Comm_f2c(mpi_comm_id);
+    MPI_Comm_size(mpi_comm, &comm_size);
+    MPI_Comm_rank(mpi_comm, &rank);
+#endif
+
+    if(node.has_child("include"))
+    {
+        int include_file_valid = 0;
+        std::string emsg = "";
+        std::string file_name = "";
+        
+        // Only want to update the node on rank 0
+        if (rank == 0) {
+            file_name = node.fetch("include").as_string();
+            node.remove_child("include");
+
+            // Determine file protocol from file extension
+            std::string curr,next;
+            std::string protocol = "json";
+            conduit::utils::rsplit_string(file_name,
+                                            ".",
+                                            curr,
+                                            next);
+            if(curr == "yaml")
+            {
+                protocol = "yaml";
+            }
+
+            // Try loading in the included path
+            try
+            {
+                conduit::Node file_node;
+                file_node.load(file_name, protocol);
+                node.update(file_node);
+                include_file_valid = 1;
+            }
+            catch(conduit::Error &e)
+            {
+                include_file_valid = 0;
+                emsg = e.message();
+            }
+        }
+
+#ifdef ASCENT_MPI_ENABLED
+        // make sure all ranks error if the parsing on rank 0 failed.
+        MPI_Bcast(&include_file_valid, 1, MPI_INT, 0, mpi_comm);
```

Comment:
If we have an error, we should broadcast the error message from rank 0 to all other ranks so their exceptions match